File: day03/star05.py
def score(c):
    s = 1 + ord(c)
    s -= ord('a') if c.islower() else (ord('A') - 26)
    return s

def find_error(line):
    line = line.strip()
    comp1 = line[:len(line)//2]
    comp2 = line[len(line)//2:]
    s1 = set(comp1)
    s2 = set(comp2)
    return s1.intersection(s2).pop()

with open('input.txt') as f:
    total = 0
    for line in f:
        total += score(find_error(line))

print(f"{total = }")

File: day03/star05.cpp
#include <fstream>
#include <string>
#include <iostream>
#include <cctype>

int score(char c) {
    return std::islower(c) ? 1 + c - 'a' : 27 + c - 'A';
}

int main() {
    std::ifstream f {"input.txt"};
    std::string line, left, right;
    
    int total {0};
    while (std::getline(f, line)) {
        left = line.substr(0, line.length()/2);
        right = line.substr(line.length()/2);
        for (auto c : left) {
            if (right.find(c) != std::string::npos) {
                total += score(c);
                break;
            }
        }
    }

    std::cout << "total = " << total << std::endl;
}